src/segmentmytiff/utils/train_model.py

```diff
@@ -0,0 +1,161 @@
+
+    if use_mlflow:
+        mlflow.end_run()
+
+
+def train(model, train_dataloader, validation_dataloader, criterion, optimizer, device, epochs = 10):
+    for epoch in tqdm(range(epochs)):
+        train_loss, train_dice = train_one_step(model, train_dataloader, optimizer, criterion, device)
+        val_loss, val_dice = validate(model, validation_dataloader, criterion, device)
+
+        print("-" * 30)
+        print(f"Training Loss EPOCH {epoch}: {train_loss:.4f}")
+        print(f"Training DICE EPOCH {epoch}: {train_dice:.4f}")
+        print("\n")
+        print(f"Validation Loss EPOCH {epoch}: {val_loss:.4f}")
+        print(f"Validation DICE EPOCH {epoch}: {val_dice:.4f}")
+        print("-" * 30)
+
+        metrics = {
+            "train_loss": train_loss,
+            "train_dc": train_dice,
+            "val_loss": val_loss,
+            "val_dc": val_dice,
+        }
+
+        mlflow.log_metrics(metrics, step=epoch, timestamp=int(round(time.time())))
+    # Saving the model
+    torch.save(model.state_dict(), 'my_checkpoint.pth')
+
+
+def validate(model, validation_dataloader, criterion, device):
+    model.eval()
+    val_running_loss = 0
+    val_running_dc = 0
+    with torch.no_grad():
+        for idx, img_mask in enumerate(tqdm(validation_dataloader, position=0, leave=True)):
+            img = img_mask[0].float().to(device)
+            mask = img_mask[1].float().to(device)
+
+            y_pred = model(img)
+            loss = criterion(y_pred, mask)
+            dc = dice_coefficient(y_pred, mask)
+
+            val_running_loss += loss.item()
+            val_running_dc += dc.item()
+
+        val_loss = val_running_loss / (idx + 1)
+        val_dc = val_running_dc / (idx + 1)
+    return val_dc, val_loss
+
+
+def train_one_step(model, train_dataloader, optimizer, criterion, device):
+    model.train()
+    train_running_loss = 0
+    train_running_dice = 0
+    for idx, img_mask in enumerate(tqdm(train_dataloader, position=0, leave=True)):
+        img = img_mask[0].float().to(device)
+        mask = img_mask[1].float().to(device)
+
+        y_pred = model(img)
+        optimizer.zero_grad()
+
+        dice = dice_coefficient(y_pred, mask)
+        loss = criterion(y_pred, mask)
+
+        train_running_loss += loss.item()
+        train_running_dice += dice.item()
+
+        loss.backward()
+        optimizer.step()
+    train_loss = train_running_loss / (idx + 1)
+    train_dice = train_running_dice / (idx + 1)
+    return train_loss, train_dice,
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Process input and output TIFF files.")
+    parser.add_argument('-r', '--root', type=Path, required=True, help='Root to the dataset')
+    parser.add_argument('--no_mlflow', action='store_true', help='Flag for enabling or disabling MLflow')
+    parser.add_argument('--train_set_limit', type=int, default=None, help='Limit for the size of the train set (default: no limit)')
```

Comment:
maybe this args parser could be define in a separate file and import it to any training script.

Reply:
Good idea, but I'm not sure whether there will be enough time in the project to actually come up with another training script.